lib/amo-client.js

```diff
@@ -110,7 +110,12 @@ Client.prototype.waitForSignedAddon = function(statusUrl, opt) {
     function checkSignedStatus() {
       self.get({url: statusUrl}).then(function(result) {
         var data = result[1];
-        if (data.processed) {
+
+        var failedValidation = !data.valid;
+        // The add-on passes validation and all files have been created.
+        var passedValidation = data.valid && data.reviewed;
+
+        if (data.processed && (failedValidation || passedValidation)) {
```

Comment:
It would need to be valid to evaluate the right hand of the or so it isn't needed in that expression. There could be a point in the future when valid is `false` and it is reviewed (we currently support submissions with a validation timeout on the frontend) so I wouldn't bank on that.